gst/nnstreamer/include/nnstreamer_plugin_api_trainer.h

```diff
@@ -0,0 +1,144 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * NNStreamer API for Tensor_Trainer Sub-Plugins
+ * Copyright (C) 2022 Hyunil Park <hyunil46.park@samsung.com>
+ */
+/**
+ * @file  nnstreamer_plugin_api_filter.h
+ * @date  1 Dec 2022
+ * @brief Mandatory APIs for NNStreamer Trainer sub-plugins (No External Dependencies)
+ * @see https://github.com/nnstreamer/nnstreamer
+ * @author  Hyunil Park <hyunil46.park@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+#ifndef __NNS_PLUGIN_API_TRAINER_H__
+#define __NNS_PLUGIN_API_TRAINER_H__
+
+#include "tensor_typedef.h"
+
+#define GST_TENSOR_TRAINER_FRAMEWORK_BASE (0xDEAFDEAD00000000ULL)
+#define GST_TENSOR_TRAINER_FRAMEWORK_V1 (GST_TENSOR_TRAINER_FRAMEWORK_BASE | 0x10000ULL)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief GstTensorTrainer's properties for neural network framework (internal data structure)
+ *
+ * Internal data of GstTensorTrainer required by tensor_trainer's custom subplugin.
+ */
+typedef struct _GstTensorTrainerProperties
+{
+  GstTensorsInfo input_meta;    /**< configured input tensor info */
+  const char *model_config;    /**< The configuration file path for creating model */
+  const char *model_save_path;    /**< The file path to save the model */
+  int64_t num_inputs;    /**< The number of input lists, the input is where framework receive the features to train the model, num_inputs indicates how many inputs there are. */
+  int64_t num_labels;    /**< The number of label lists, the label is where framework receive the class to train the model, num_labels indicates how many labels there are. */
```

Comment:
Is it common for all?

Reply:
Thank you.
I can't say for sure now. This structure is still under development.
However, since this PR is related to nntrainer and nnstreamer tensor_trainer, merge is needed. That's why [myungjoo](https://github.com/myungjoo) approved.
This structure is a temporay code. And it seems that a lot will change in the future.
I'll make a new PR when I need to fix it.

Reply:
So are you saying ```num_labels``` is required to merge this PR? If not, you are trying to push the unnecessary line to upstream. It is better to leave the NECESSARY lines to be required by nntrainer.

Reply:
Sorry. I thought what you mentioned was not about ```'num_labels```, but about all member variables.
We also need ```'num_labels```.
we can think of some case.
There can be more than one label for multiple features.
```
Feature = N, label = 2
feature 1----|
feature 2----|                             |--> label1
feature 3----| ---------> [DNN]----|--> label2 
...
feature N---|
```
For TTS support, nntrainer has received multiple labels.
Thank you.